src/GHC/TypeLits/Extra/Solver/Operations.hs

```diff
@@ -138,6 +156,31 @@ reifyEOP defs (LCM x y)  = mkTyConApp (lcmTyCon defs)  [reifyEOP defs x
 reifyEOP defs (Exp x y)  = mkTyConApp typeNatExpTyCon  [reifyEOP defs x
                                                        ,reifyEOP defs y]
 
+mergePlus :: ExtraDefs -> ExtraOp -> ExtraOp -> NormaliseResult
+mergePlus _ (Max a b) y = (Max (Plus a y) (Plus b y), Normalised)
+mergePlus _ x (Max a b) = (Max (Plus x a) (Plus x b), Normalised)
+mergePlus _ (Min a b) y = (Min (Plus a y) (Plus b y), Normalised)
+mergePlus _ x (Min a b) = (Min (Plus x a) (Plus x b), Normalised)
+-- Give a canonical ordering so we can pretend like we have commutativity.
+mergePlus _ x y = (case x <= y of
+                     True -> Plus x y
+                     False-> Plus y x, Untouched)
+
+mergeSub :: ExtraDefs -> ExtraOp -> ExtraOp -> NormaliseResult
+mergeSub _ (Max a b) y = (Max (Sub a y) (Sub b y), Normalised)
```

Comment:
Subtraction is tricky, also because type families are in a way "strict". So technically `Max 3 4 - 4` would reduce, while `Max (3-4) (4-4)`  would/should not because `3-4` will become "stuck". Perhaps we can just skip subtraction for now.

Reply:
I'm not sure it helps my original use case if this simplification doesn't happen, unfortunately. Maybe a flag for subtraction like natnormalise has?

Reply:
I guess we should be fine actually. I don't believe it can lead to the unsound behavior described in: https://hackage.haskell.org/package/ghc-typelits-natnormalise-0.7.6/docs/GHC-TypeLits-Normalise.html